ai/spaced-repetition/main.py

```diff
@@ -0,0 +1,63 @@
+class SM2:
+    def __init__(self):
+        self.ease_factor = 2.5  # Initial ease factor
+        # Initial interval (days), inter-repetition interval after the n-th repetition
+        self.interval = 1
+        self.repetitions = 0    # Number of repetitions
+
+    def update(self, quality):
+        """
+        Parameters:
+        quality (int): The quality of the response (0 to 5).
+                       0: complete blackout,
+                       1: incorrect response but remembered,
+                       2: incorrect response but easy to recall,
+                       3: correct response but hard to recall,
+                       4: correct response with hesitation,
+                       5: perfect response.
+        """
+        if quality < 3:
+            self.repetitions = 0
+            self.interval = 1
+        else:
+            if self.repetitions == 0:
+                self.interval = 1
+            elif self.repetitions == 1:
+                self.interval = 6
+            else:
+                self.interval *= self.ease_factor
+
+            self.repetitions += 1
+
+        # EF’:=EF+(0.1-(5-q)*(0.08+(5-q)*0.02))
+        self.ease_factor = max(
+            1.3, self.ease_factor + 0.1 - (5 - quality) * (0.08 + (5 - quality) * 0.02))
```

Comment:
Little confused why there is a `max` function ? 

Shouldn't it be: `self.ease_factor = self.ease_factor + ( 0.1 - (5 - quality) * (0.08 + (5 - quality) * 0.02) )`?

Reply:
Nvm: "If EF is less than 1.3 then let EF be 1.3."